Modules/RatMama.py

```diff
@@ -0,0 +1,176 @@
+    o2_status: bool = result.group("o2") == "OK"  # false is CR
+    lang_code: str = result.group("language_code")
+    nickname: Optional[str] = result.group("nick")
+
+    exist_rescue: Rescue = board.find_by_name(client_name)
+
+    if exist_rescue:
+        # we got a case already!
+        await ctx.reply(f"{client_name} has reconnected! Case #{exist_rescue.board_index}")
+        # now let's make it more visible if stuff changed
+        diff_response: str = ""
+        if system_name.casefold() != exist_rescue.system.casefold():
+            diff_response += f"System changed! "
+
+        if platform_name.casefold() != exist_rescue.platform.name.casefold():
+            diff_response += "Platform changed! "
+
+        if not o2_status != exist_rescue.code_red:
+            diff_response += "O2 Status changed!" if o2_status else\
+                "O2 Status changed, it is now CODE RED!"
+
+        if not diff_response == "":
+            await ctx.reply(diff_response)
+
+    else:
+        platform = None
+
+        if platform_name.casefold() in ("pc", "ps", "xb"):
+            platform = Platforms[platform_name.upper()]
+        else:
+            LOG.warning(f"Got unknown platform from RatMama: {platform_name}")
+
+        # no case for that name, we have to make our own
+        rescue: Rescue = Rescue(client=client_name, system=system_name, irc_nickname=nickname,
+                                code_red=not o2_status, lang_id=lang_code, platform=platform)
+
+        board.append(rescue, overwrite=False)
+        index = board.find_by_name(client=client_name).board_index
+        await ctx.reply(f"RATSIGNAL - CMDR {client_name} - "
+                        f"Reported System: {system_name} (distance to be implemented) - "
+                        f"Platform: {platform_name} - "
+                        f"O2: {'OK' if o2_status else 'NOT OK'} - "
+                        f"Language: {result.group('full_language')}"
+                        f" (Case #{index}) ({platform_name.upper()}_SIGNAL)"
+                        )
+
+
+@rule("ratsignal", case_sensitive=False, full_message=False, pass_match=False, prefixless=True)
+async def handle_selfissued_ratsignal(ctx: Context):
+    message: str = ctx.words_eol[0]
+    # the ratsignal is nothing we are interested anymore
+    message = re.sub("ratsignal", "", message, flags=re.I)
+
+    for rescue in board.rescues.values():
+        if rescue.irc_nickname.casefold() == ctx.user.nickname.casefold():
+            ctx.reply("You already sent a signal, please be patient while a dispatch is underway.")
+            return
+
+    sep: chr = None
+    if ',' in message:
+        sep = ','
+    elif ';' in message:
+        sep = ';'
+    elif '|' in message:
+        sep = '|'
+    elif '-' in message:
+        sep = '-'
+
+    if not sep:
+        board.append(Rescue(irc_nickname=ctx.user.nickname, client=ctx.user.nickname))
+        index = board.find_by_name(ctx.user.nickname)
```

Comment:
Coverage suggests there is no test for this case.

Further, it appears there's a small error here, as you're setting `index` to be the `Rescue` object itself, rather than its board index. I'd also like to reiterate my previous comment about searching the board for a rescue we just created.

Reply:
This is still the case. Why is the conversation resolved?